torch/profiler/_pattern_matcher.py

```diff
@@ -176,7 +178,24 @@ def match(self, event):
         # TODO: We should also check tensor identities
         if event.name() != "aten::to":
             return False
+        to_event = event
+        if not event.children:
+            return False
+        event = event.children[-1]
+        if event.name() != "aten::_to_copy":
+            return False
+        if not event.children:
+            return False
+        event = event.children[-1]
+        if event.name() != "aten::copy_":
+            return False
+        if not event.children:
+            return False
+        event = event.children[0]
+        if event.name() != "cudaMemcpyAsync":
```

Comment:
This is probably why rocm CI was failing.  Exact string matching on CUDA APIs.

Reply:
👍 Thanks for pointing it out!

Reply:
```suggestion
        if event.name() not in ("cudaMemcpyAsync", "hipMemcpyWithStream"):
```

If you rework your PR, the above should fix ROCm CI.